onnxruntime/core/providers/partitioning_utils.cc

```diff
@@ -160,24 +160,59 @@ std::vector<std::vector<const Node*>> CreateSupportedPartitionNodeGroups(
         node.GetExecutionProviderType().empty() &&  // a node that is already assigned to an EP is unsupported
         is_node_supported_fn(node);
 
+    // TODO: special handling for a node in a qdq group
+    nnapi::QDQSupportHelper qdq_support_helper = nnapi::QDQSupportHelper(nnapi::CreateSelectors(), graph_viewer);
+    const auto is_qdq_node = qdq_support_helper.IsNodeInQDQGroup(node);
+
     if (!is_node_supported && Contains(supported_group_border, &node)) {
       // an unsupported node on the border will be processed after the current partition node group
-      nodes_to_process_with_next_group.push_back(&node);
-      continue;
+      // for an unsupported node in a qdq group, we just don't process it
+      if (!is_qdq_node) {
+        nodes_to_process_with_next_group.push_back(&node);
+        continue;
+      }
     }
 
     if (is_node_supported) {
-      // add node to the partition node group
-      supported_group.push_back(&node);
-
-      // remove node from the border and add its outputs to the border
-      supported_group_border.erase(&node);
-
-      std::for_each(
-          node.OutputNodesBegin(), node.OutputNodesEnd(),
-          [&supported_group_border](const Node& output) {
-            supported_group_border.insert(&output);
-          });
+      if (is_qdq_node) {
```

Comment:
It is apparently not ideal to add special handlings for qdq case in partitioning utils..
But just to get an idea, what we want to do here is to treat a qdq_node_group "as a whole"
- if a node is a qdq node and it's supported, check if all dq/q/target nodes are supported. Only if all of them are supported, we add the qdq nodes into supported_group altogether. (Here for now, we only process when we hit a supported target node in a qdq group, if we meet a supported non-target node, we do nothing.)
- if a node is a qdq node and it's not supported, we simply don't add it to nodes_to_process.

There exists re-visit node issue here.  Also maybe the idea of making"a wrapper Node class" which contains more info and pass that into is_node_supported_fn can be a better and more general to deal with this.